o11y/httpmetrics/provider.go

```diff
@@ -105,18 +106,12 @@ func (m *Provider) Count(n string, v int64, t []string, rate float64) error {
 }
 
 func (m *Provider) Close() error {
-	m.stopMu.Lock()
-	defer m.stopMu.Unlock()
-
-	if m.stop != nil {
+	m.stopOnce.Do(func() {
 		close(m.stop)
-		m.stop = nil
```

Comment:
this is the race since the mutex does not guard other uses of `stop` - it only protects agains concurrent 'Close' ..

it does not protect against Close - whilst `case <-m.stop` is being called in the loop